actions/teacher/create.ts

```diff
@@ -1,25 +1,43 @@
 'use server'
 
-import { Teacher } from "@/models/Teacher";
-import dbConnect from "@/lib/dbConnect";
-import mongoose from "mongoose";
+import dbConnect from '@/lib/dbConnect'
+import { ITeacher, Teacher } from '@/models/Teacher'
+import { User } from '@/models/User'
+import { createUser } from '../user/create'
 
-interface CreateTeacherInput {
-  user: string;
-}
+// Acción para crear un profesor y asociarlo a un área
+export async function createTeacher(formData: FormData): Promise<ITeacher> {
+  await dbConnect()
 
-export async function createTeacher(data: CreateTeacherInput) {
-  await dbConnect();
+  const email = formData.get('email')?.toString() || ''
+  const areaId = formData.get('area')?.toString()
 
-  if (!mongoose.Types.ObjectId.isValid(data.user)) {
-    throw new Error("ID de usuario inválido");
+  if (!email || !areaId) {
+    throw new Error('Email y área son obligatorios')
   }
 
-  const newTeacher = new Teacher({
-    user: new mongoose.Types.ObjectId(data.user),
-  });
+  // 1. Buscar si el usuario ya existe
+  let user = await User.findOne({ email })
+
+  if (!user) {
+    // 2. Crear nuevo usuario si no existe
+    user = await createUser(formData)
+    if (!user || !user._id) {
+      throw new Error('No se pudo crear el usuario')
+    }
+  }
+
+  // 3. Verificar si ya existe un Teacher para ese usuario
+  const existingTeacher = await Teacher.findOne({ user: user._id })
+  if (existingTeacher) {
+    throw new Error('Este usuario ya está registrado como profesor')
+  }
 
-  const savedTeacher = await Teacher.create(newTeacher);
+  // 4. Crear el nuevo Teacher con el área asociada
+  const teacher = await Teacher.create({
+    user: user._id,
+    area: areaId,
+  })
```

Comment:
Quizas podemos agrega
```suggestion
  await addTeacherToArea(areaId, teacherId)
```